### PR TITLE
WR 1582 french banner editable in cms

### PR DESCRIPTION
**Pull request by rjkuyvenhoven**
This branch requires some manual configuration due to "config_ignore" settings.

Go to /admin/structure/block
Click 'Place block' button next to 'Hidden (For Blocks and Paragraphs'
Find 'WorkBC Switch Language Block - WorkBC' and click 'Place block'
Ensure machine name is 'workbcswitchlanguageblock'
Uncheck 'Display title' in block configuration
Check 'Page' under 'Content type' tab
Click 'Save blocks'

### CODE REVIEW

**Comment by infojunkie** [comment]
Basic testing worked well, thanks. Some comments:

1. Make sure this system works for WorkBC Centres out of the box (i.e. the new block is enabled for WorkBC Centres and works as expected there)

2. Provide a migration path for the existing pages that are hard-coded to contain French links (as evidenced by their custom blocks in the attached screenshot). For example:
  - Disable the shown blocks
  - Write a migration for these pages to populate their multilingual fields
  - OR Write a recipe on adding links to these page explicitly
screenshot-workbc docker localhost_8000-2023 06 15-17_57_54

3. Make sure that the theming for the links is consistent. Attached is the same link rendered for pages with and without hero images:
screenshot-workbc docker localhost_8000-2023 06 15-17_57_33
screenshot-workbc docker localhost_8000-2023 06 15-17_56_26

**Review by infojunkie — CHANGES_REQUESTED** [review]
Some changes needed as per my comment above

File: src/web/modules/custom/workbc_custom/src/Plugin/Block/SwitchLanguageBlock.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+use Drupal\Core\Url;
+use Drupal\Core\Link;
+/**
+ * Provides a WorkBC Switch language link Block for content that has a value in
+ * the Multilingual centre field.
+ *
+ * @Block(
+ *   id = "switch_language_block",
+ *   admin_label = @Translation("WorkBC switch language block"),
+ *   category = @Translation("WorkBC"),
```
Comment (infojunkie) [inline src/web/modules/custom/workbc_custom/src/Plugin/Block/SwitchLanguageBlock.php:17]: Please call this "WorkBC Custom" like other blocks in this module.

Comment (rjkuyvenhoven) [reply src/web/modules/custom/workbc_custom/src/Plugin/Block/SwitchLanguageBlock.php:17]: I can make this change. However, none of the blocks in this module use that naming convention.

Comment (infojunkie) [reply src/web/modules/custom/workbc_custom/src/Plugin/Block/SwitchLanguageBlock.php:17]: I was confused by some blocks missing the category "WorkBC". I pushed a commit to fix that.

File: src/web/themes/custom/workbc/templates/content/node--page--full.html.twig [hunk 2]
```diff
@@ -110,6 +111,13 @@
...
 
+{# Switch Language Link #}
+{% if switch_language_link %}
+  <div class="fr-link">
+    {{ switch_language_link }}
+  </div>
+{% endif %}
+
 {# French Link #}
 {% include '@workbc/includes/french-link.html.twig' %}
```
Comment (infojunkie) [inline src/web/themes/custom/workbc/templates/content/node--page--full.html.twig:122]: This should no longer be used

**Comment by rjkuyvenhoven** [comment]
To enable multilingual support for WorkBC Centre content type:

Go to /admin/structure/block
Click 'Configure' for WorkBC switch language block
Check 'WorkBC Centre' under 'Content type' tab
Click 'Save blocks'


- WorkBC Locations now work out of the box

- Removing original French Link section in Page twig file resolves the inconsistent link rendering b/w Hero Image and no Hero Image pages.  NOTE: This will result in links to French pages to disappear.  The appropriate pages will need to be edited to add the appropriate links.  


The following pages and their corresponding French pages need to be edited to use the new Multilingual functionality.

- Contact Us
- Online Employment Services
- Employment Services
- Who Should Visit Workbc Centre
- Self Serve Services
- Workbc Centre Locations


For each page:

- Edit English page
- Search and select corresponding French page in Multilingual Page -> URL field
- Enter appropriate text in Multilingual Page -> Link text field
- Save page
- Edit corresponding French page
- Search and select corresponding English page in Multilingual Page -> URL field
- Enter appropriate text in Multilingual Page -> Link text field
- Remove "Service in English" button/link from Body field (if desired).
- Save page

Both pages should now display a link to the other page.

**Review by infojunkie — APPROVED** [review]
Lgtm. Please adjust the 6 manual pages yourself on DEV, and include the config + migration recipe for TEST and PROD in the ticket. Thanks!

**Pull request merged**